plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/BaseDeltaLakeRegisterTableProcedureTest.java

```diff
@@ -0,0 +1,273 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.deltalake;
+
+import io.trino.hdfs.HdfsContext;
+import io.trino.plugin.hive.metastore.HiveMetastore;
+import io.trino.testing.AbstractTestQueryFramework;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
+
+import static com.google.common.io.MoreFiles.deleteDirectoryContents;
+import static com.google.common.io.MoreFiles.deleteRecursively;
+import static com.google.common.io.RecursiveDeleteOption.ALLOW_INSECURE;
+import static io.trino.plugin.deltalake.transactionlog.TransactionLogUtil.getTransactionLogDir;
+import static io.trino.plugin.deltalake.transactionlog.TransactionLogUtil.getTransactionLogJsonEntryPath;
+import static io.trino.testing.sql.TestTable.randomTableSuffix;
+import static java.lang.String.format;
+import static org.assertj.core.api.AssertionsForInterfaceTypes.assertThat;
+
+public abstract class BaseDeltaLakeRegisterTableProcedureTest
+        extends AbstractTestQueryFramework
+{
+    protected static final String SCHEMA = "test_tables_register_table_" + randomTableSuffix();
+    protected static final String CATALOG_NAME = "delta_register_table";
+    protected File metastoreDir;
+    protected HiveMetastore metastore;
+    protected HdfsContext hdfsContext;
+
+    @Test
+    public void testRegisterTableWithTableLocation()
+    {
+        String tableName = "test_register_table_with_table_location_" + randomTableSuffix();
+
+        assertQuerySucceeds(format("CREATE TABLE %s AS SELECT 1 as a, 'INDIA' as b, true as c", tableName));
+        assertThat(query(format("SELECT * FROM %s", tableName)))
+                .matches("VALUES ROW(INT '1', VARCHAR 'INDIA', BOOLEAN 'true')");
+
+        String tableLocation = getTableLocation(tableName);
+
+        // Drop table from hive metastore and use the same table name to register again with the transaction logs
+        dropTableFromMetastore(tableName);
+
+        assertQuerySucceeds(format("CALL %s.system.register_table('%s', '%s', '%s')", CATALOG_NAME, SCHEMA, tableName, tableLocation));
```

Comment:
We should verify that the content of the table is is still there to make sure that the registration was indeed effective.

Reply:
Addressed!